mm2src/lp_swap.rs

```diff
@@ -264,10 +267,44 @@ async fn recv_swap_msg<T>(
 /// in order to give different and/or heavy communication channels a chance.
 const BASIC_COMM_TIMEOUT: u64 = 90;
 
+#[cfg(not(feature = "custom-swap-locktime"))]
 /// Default atomic swap payment locktime, in seconds.
 /// Maker sends payment with LOCKTIME * 2
 /// Taker sends payment with LOCKTIME
-pub const PAYMENT_LOCKTIME: u64 = 3600 * 2 + 300 * 2;
+const PAYMENT_LOCKTIME: u64 = 3600 * 2 + 300 * 2;
+
+#[cfg(feature = "custom-swap-locktime")]
+/// Reads `payment_locktime` from cli or MM2.json configuration and returns it.
+/// Returns 900 when `payment_locktime` is invalid or not provided.
+pub fn get_payment_locktime_from_mm2config() -> u64 {
+    let args: Vec<OsString> = env::args_os().collect();
+    let first_arg = args.get(1).and_then(|arg| arg.to_str());
+    let conf = super::get_mm2config(first_arg).unwrap();
+
+    match conf["payment_locktime"].as_u64() {
+        Some(lt) => lt,
+        None => {
+            let def = 900;
+            warn!(
+                "payment_locktime is either invalid type or not provided in the configuration or
+                MM2.json file. payment_locktime will be proceeded as {} seconds.",
+                def
+            );
+            def
+        },
+    }
+}
+
+#[inline]
+/// Returns `PAYMENT_LOCKTIME` or result of `get_payment_locktime_from_mm2config()`
+/// depended on the activation of `custom-swap-locktime` feature.
+pub fn get_payment_locktime() -> u64 {
+    #[cfg(not(feature = "custom-swap-locktime"))]
+    return PAYMENT_LOCKTIME;
+    #[cfg(feature = "custom-swap-locktime")]
+    get_payment_locktime_from_mm2config()
```

Comment:
Even for tests, it may not be optimal to read the config and parse it several times per swap.
Also the field value can be changed between `get_payment_locktime` calls.
What do you think about adding a lazy static variable?

```rust
#[cfg(feature = "custom-swap-locktime")]
azy_static! { static mut PAYMENT_LOCKTIME: u64 = get_payment_locktime_from_mm2config(); }
```

Reply:
> Even for tests, it may not be optimal to read the config and parse it several times per swap. Also the field value can be changed between `get_payment_locktime` calls. What do you think about adding a lazy static variable?
> 
> ```rust
> #[cfg(feature = "custom-swap-locktime")]
> azy_static! { static mut PAYMENT_LOCKTIME: u64 = get_payment_locktime_from_mm2config(); }
> ```

Thank you for suggestion, its' much better now.